File: chroma-league/src/main/java/net/booone/chromaleague/tasks/FetchActivePlayerTask.java
package net.booone.chromaleague.tasks;

import net.booone.chromaleague.rest.http.LeagueHttpClients;
import net.booone.chromaleague.rest.http.handlers.ActivePlayerResponseHandler;
import net.booone.chromaleague.state.RunningState;

public class FetchActivePlayerTask implements Runnable {
    public static final String URL = "https://127.0.0.1:2999/liveclientdata/activeplayer";
    private final ActivePlayerResponseHandler activePlayerResponseHandler = new ActivePlayerResponseHandler();

    @Override
    public void run() {
        LeagueHttpClients.getNonBlockingResponse(URL, activePlayerResponseHandler)
                .ifPresent(activePlayer -> RunningState.getGameState().setActivePlayer(activePlayer));
    }
}

File: src/tasks/FetchActivePlayerTask.ts
import {getNonBlockingResponse} from '../rest/http/LeagueHttpClients.js';
import type {ActivePlayer} from '../rest/activeplayer/ActivePlayer.js';
import {RunningState} from '../state/RunningState.js';

const URL = 'https://127.0.0.1:2999/liveclientdata/activeplayer';

export async function fetchActivePlayerTask(): Promise<void> {
    const activePlayer = await getNonBlockingResponse<ActivePlayer>(URL);
    if (activePlayer) {
        RunningState.getGameState().activePlayer = activePlayer;
    }
}
